### PR TITLE
115 multi publish config

### PR DESCRIPTION
**Pull request by eiabea**

### CODE REVIEW

File: helper/project.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import shelljs, { ExecOutputReturnValue } from "shelljs";
 import { IIntegrationLink, IJiraLink, IProject, IProjectMeta, IRecord } from "../interfaces";
 import { GitNoOriginError, GitNoUrlError, GitRemoteError, GitNoRepoError, RECORD_TYPES } from "../types";
+import { IMultipieLink } from '../interfaces/index';
```
Comment (lombad) [inline helper/project.ts]: add `IMultipieLink` to previous import.

File: helper/project.ts [hunk 2]
```diff
@@ -320,23 +321,31 @@ export class ProjectHelper {
...
-        default:
-          LogHelper.error("✗ Invalid link type");
-          break;
+      for (const link of links) {
+        switch (link.linkType) {
+          case "Jira":
+            const migratedJiraLink: IJiraLink = link as IJiraLink;
+            migratedJiraLink.projectName = to.name;
+
+            await this.fileHelper.addOrUpdateLink(migratedJiraLink);
```
Comment (lombad) [inline helper/project.ts]: Move the `addOrUpdateLink` call outside of switch case and init variable as `IIntegrationLink`. Also change `addOrUpdateLink` to get only `IIntegrationLink` types.

Comment (lombad) [reply helper/project.ts]: see `file.ts` (aka FileHelper)

```
  public addOrUpdateLink = async (link: IIntegrationLink | IJiraLink | IMultipieLink): Promise<IConfigFile> => {
     ....
  }
```

File: helper/question.ts [hunk 3]
```diff
@@ -354,6 +354,7 @@ export class QuestionHelper {
     return choice.choice;
   }
 
+  // TODO rename to more generic function name
```
Comment (lombad) [inline helper/question.ts]: Maybe resolve this TODO ?

File: test/unit/file.helper.test.ts [hunk 4]
```diff
@@ -763,6 +763,38 @@ describe("FileHelper", function () {
       assert.isTrue(getConfigObjectStub.calledOnce);
     });
 
+    it("should add different type of link link to config file", async function () {
```
Comment (lombad) [inline test/unit/file.helper.test.ts]: `link link` ?

Comment (lombad) [reply test/unit/file.helper.test.ts]: Do you mean `link type`

File: test/unit/file.helper.test.ts [hunk 5]
```diff
@@ -799,6 +831,64 @@ describe("FileHelper", function () {
...
+      });
+
+      const updatedConfigFile: IConfigFile = await instance.addOrUpdateLink({
+        endpoint: "http://test.com/api",
+        hash: "1234asdf",
+        key: "test",
+        linkType: "mock",
+        projectName: "mocked",
+        username: "updated",
+      });
```
Comment (lombad) [inline test/unit/file.helper.test.ts]: add `as IJiraLink` type or which one was used.

File: test/unit/file.helper.test.ts [hunk 6]
```diff
@@ -799,6 +831,64 @@ describe("FileHelper", function () {
...
+
+      expect(updatedConfigFile.links.length).to.eq(2);
+      expect(updatedConfigFile).to.deep.equal({
+        created: 1234,
+        gitRepo: "ssh://mocked",
+        links: [
+          {
+            linkType: "fake",
+            projectName: "mocked",
+          },
```
Comment (lombad) [inline test/unit/file.helper.test.ts]: add `type` of Link `IIntegrationLink | IJiraLink | IMultipieLink`

Comment (lombad) [reply test/unit/file.helper.test.ts]: Pls use that in all cases a type is used.

Comment (lombad) [reply test/unit/file.helper.test.ts]: Hence any changes to the types would be visible in the Tests for later usage purposes.

File: helper/project.ts (hunk 2 above)
Comment (eiabea) [reply helper/project.ts]: I would postpone this refactoring to the `class` vs. `interface` refactor issue

File: test/unit/file.helper.test.ts (hunk 5 above)
Comment (eiabea) [reply test/unit/file.helper.test.ts]: I changed it in this particular case, but i will add this as an additional bullet point for the refactoring issue

File: test/unit/file.helper.test.ts (hunk 6 above)
Comment (eiabea) [reply test/unit/file.helper.test.ts]: Will be added to the refactoring issue

**Comment by eiabea** [comment]
@lombad Refactoring issue: GITTT-118

File: helper/project.ts (hunk 2 above)
Comment (lombad) [reply helper/project.ts]: Accepted

File: test/unit/file.helper.test.ts (hunk 5 above)
Comment (lombad) [reply test/unit/file.helper.test.ts]: changed it in a519adc5da2e7d8e272ca7e8fadfda713da62b81

File: test/unit/file.helper.test.ts (hunk 6 above)
Comment (lombad) [reply test/unit/file.helper.test.ts]: chaged it in a519adc5da2e7d8e272ca7e8fadfda713da62b81

File: helper/file.ts [hunk 7]
```diff
@@ -75,7 +75,7 @@ export class FileHelper {
     const cleanLinks: IIntegrationLink[] = configObject.links.filter((li: IIntegrationLink) => {
       // TODO linkType can be taken from class
```
[the remark this answers is not in the document]
Comment (lombad) [reply helper/file.ts]: will be changed with GITTT-118

**Pull request merged**